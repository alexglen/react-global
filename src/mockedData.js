export const mockedData = [
  {
    title: 'Green mile',
    genre: 'Drama, Fantasy',
    releaseDate: 1994,
    id: '1',
    img: 'https://pbs.twimg.com/media/ELeJbCYXkAE18WI.jpg',
    rating: 4.9,
    duration: '120',
    description:
      'The Green Mile is a 1999 American fantasy drama film written and directed by Frank Darabont and based on Stephen Kings 1996 novel of the same name. It stars Tom Hanks as a death row corrections officer during the Great Depression who witnesses supernatural events that occur after an enigmatic inmate (Michael Clarke Duncan) is brought to his facility. David Morse, Bonnie Hunt, Doug Hutchison and James Cromwell appear in supporting roles.The film premiered on December 10, 1999, in the United States to positive reviews from critics, who praised its visual style and performances. It was a commercial success, grossing $286 million from its $60 million budget, and was nominated for four Academy Awards: Best Picture, Best Supporting Actor for Duncan, Best Sound and Best Screenplay Based on Material Previously Produced or Published.',
  },
  {
    title: 'Interstellar',
    genre: 'Drama, Fantasy, Adventures',
    releaseDate: 2014,
    id: '2',
    rating: 4.8,
    duration: '124',
    img:
      'https://mir-s3-cdn-cf.behance.net/project_modules/1400_opt_1/b0f70f29979965.560d298403bc7.jpg',
    description:
      'Brothers Christopher and Jonathan Nolan wrote the screenplay, which had its origins in a script Jonathan developed in 2007. Christopher produced Interstellar with his wife, Emma Thomas, through their production company Syncopy, and with Lynda Obst through Lynda Obst Productions. Caltech theoretical physicist and 2017 Nobel laureate in Physics[4] Kip Thorne was an executive producer, acted as scientific consultant, and wrote a tie-in book, The Science of Interstellar. Paramount Pictures, Warner Bros. Pictures, and Skydance Productions co-financed the film. Cinematographer Hoyte van Hoytema shot it on 35 mm in the Panavision anamorphic format and IMAX 70 mm. Principal photography began in late 2013 and took place in Alberta (Canada), Iceland and Los Angeles. Interstellar uses extensive practical and miniature effects and the company Double Negative created additional digital effects.',
  },
  {
    title: 'Forrest Gump',
    genre: 'Drama, Comedy',
    releaseDate: 1994,
    rating: 4.7,
    id: '3',
    duration: '110',
    img: 'https://pbs.twimg.com/media/EK7Tf-IXkAErCgW.jpg',
    description:
      'Forrest Gump is a 1994 American romantic comedy-drama film directed by Robert Zemeckis and written by Eric Roth. It is based on the 1986 novel of the same name by Winston Groom and stars Tom Hanks, Robin Wright, Gary Sinise, Mykelti Williamson and Sally Field. The story depicts several decades in the life of Forrest Gump (Hanks), a slow-witted but kind-hearted man from Alabama who witnesses and unwittingly influences several defining historical events in the 20th century United States. The film differs substantially from the novel. Principal photography took place between August and December 1993, mainly in Georgia, North Carolina and South Carolina. Extensive visual effects were used to incorporate Hanks into archived footage and to develop other scenes. The soundtrack features songs reflecting the different periods seen in the film.',
  },
  {
    title: 'Shutter Island',
    genre: 'Drama, Thriller',
    releaseDate: 2009,
    rating: 4.6,
    duration: '134',
    id: '4',
    img: 'https://pbs.twimg.com/media/EHb-eADX4AAP_6T.jpg',
    description:
      'Shutter Island is a 2010 American neo-noir psychological thriller film directed by Martin Scorsese and written by Laeta Kalogridis, based on Dennis Lehanes 2003 novel of the same name. Leonardo DiCaprio stars as Deputy U.S. Marshal Edward "Teddy" Daniels, who is investigating a psychiatric facility on Shutter Island after one of the patients goes missing. Mark Ruffalo plays his partner and fellow deputy marshal, Ben Kingsley is the facilitys lead psychiatrist, Max von Sydow is a German doctor, and Michelle Williams is Danielss wife. Released on February 19, 2010, the film received mostly positive reviews from critics, was chosen by National Board of Review as one of the top ten films of 2010, and grossed over $299 million worldwide.',
  },
  {
    title: 'The Matrix',
    genre: 'Fantasy, Thriller',
    rating: 4.5,
    releaseDate: 1999,
    duration: '150',
    id: '5',
    img: 'https://cinelog.net/wp-content/uploads/2020/04/aOIuZAjPaRIE6CMzbazvcHuHXDc-768x1152.jpg',
    description:
      'The Matrix is an American media franchise created by writers-directors the Wachowskis and producer Joel Silver. The series consists of a three movies, beginning with The Matrix (1999) and continuing with two sequels, The Matrix Reloaded and The Matrix Revolutions (both in 2003), all written and directed by the Wachowskis and produced by Joel Silver. The franchise is owned by Warner Bros., which distributed the films along with Village Roadshow Pictures. The latter, along with Silver Pictures, are the two production companies that worked on all three films.The series features a cyberpunk story of the technological fall of mankind, in which the creation of artificial intelligence led the way to a race of self-aware machines that imprisoned mankind in a virtual reality system—the Matrix—',
  },
  {
    title: 'The brother',
    genre: 'Drama, Thriller',
    rating: 4.3,
    releaseDate: 1997,
    duration: '112',
    id: '6',
    img: 'https://i10.fotocdn.net/s124/a80214fc2d561294/user_xl/2825091898.jpg',
    description:
      'Brother (Russian: Брат, translit. Brat) is a 1997 Russian neo-noir crime drama film written and directed by Aleksei Balabanov. The film stars Sergei Bodrov Jr. as Danila Bagrov, a young ex-army conscript who becomes embroiled with the Saint Petersburg mob through his criminal older brother. It appeared in the Un Certain Regard section at the 1997 Cannes Film Festival.[1]After its release on VHS in June 1997, Brother unexpectedly became one of the most commercially successful Russian films of the 1990s and quickly rose to cult film status throughout Russia. Due to the films popularity and fan demand, a sequel, Brother 2, was released in 2000.',
  },
];
