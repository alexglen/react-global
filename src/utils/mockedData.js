export const mockedData = [
  {
    title: "Green mile",
    genre: "Drama, Fantasy",
    releaseDate: 1994,
    id: "1",
    img: "https://pbs.twimg.com/media/ELeJbCYXkAE18WI.jpg",
  },
  {
    title: "Interstellar",
    genre: "Drama, Fantasy, Adventures",
    releaseDate: 2014,
    id: "2",
    img: "https://mir-s3-cdn-cf.behance.net/project_modules/1400_opt_1/b0f70f29979965.560d298403bc7.jpg",
  },
  {
    title: "Forrest Gump",
    genre: "Drama, Comedy",
    releaseDate: 1994,
    id: "3",
    img: "https://pbs.twimg.com/media/EK7Tf-IXkAErCgW.jpg",
  },
  {
    title: "Shutter Island",
    genre: "Drama, Thriller",
    releaseDate: 2009,
    id: "4",
    img: "https://pbs.twimg.com/media/EHb-eADX4AAP_6T.jpg",
  },
  {
    title: "The Matrix",
    genre: "Fantasy, Thriller",
    releaseDate: 1999,
    id: "5",
    img: "https://cinelog.net/wp-content/uploads/2020/04/aOIuZAjPaRIE6CMzbazvcHuHXDc-768x1152.jpg",
  },
  {
    title: "The brother",
    genre: "Drama, Thriller",
    releaseDate: 1997,
    id: "6",
    img: "https://i10.fotocdn.net/s124/a80214fc2d561294/user_xl/2825091898.jpg",
  },
];
