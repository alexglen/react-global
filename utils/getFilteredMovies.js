import { allMovies } from "../constants";

export const getFilteredMovies = (filter, movies) => {
  if (filter === allMovies) {
    return movies;
  } else {
    return movies.filter(({ genre }) => genre.includes(filter));
  }
};
