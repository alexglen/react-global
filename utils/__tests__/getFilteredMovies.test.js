import { getFilteredMovies } from "../getFilteredMovies";
import { mockedData } from "../mockedData";

describe("getFilteredMovies", () => {
  test.each`
    filter
    ${"all"}
    ${"Drama"}
    ${"Fantasy"}
    ${"Horror"}
  `('(filter: "$filter")', ({ filter }) => {
    expect(getFilteredMovies(filter, mockedData)).toMatchSnapshot();
  });
});
