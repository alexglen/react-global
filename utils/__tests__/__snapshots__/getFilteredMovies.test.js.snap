// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getFilteredMovies (filter: "Drama") 1`] = `
Array [
  Object {
    "description": "The Green Mile is a 1999 American fantasy drama film writtenand directed by Frank Darabont and based on Stephen Kings 1996 novel of the same name. It stars Tom Hanks as a death row corrections officer during the Great Depression who witnesses supernatural events that occur after an enigmatic inmate (Michael Clarke Duncan) is brought to his facility. David Morse, Bonnie Hunt, Doug Hutchison and James Cromwell appear in supporting roles.The film premiered on December 10, 1999, in the United States to positive reviews from critics, who praised its visual style and performances. It was a commercial success, grossing $286 million from its $60 million budget, and was nominated for four Academy Awards: Best Picture, Best Supporting Actor for Duncan, Best Sound and Best Screenplay Based on Material Previously Produced or Published.",
    "duration": "120",
    "genre": "Drama, Fantasy",
    "id": "-MWJmKyNgxg1U3WKo2MF",
    "img": "https://pbs.twimg.com/media/ELeJbCYXkAE18WI.jpg",
    "rating": "4.9",
    "releaseDate": "1994",
    "title": "Green mile",
  },
]
`;

exports[`getFilteredMovies (filter: "Fantasy") 1`] = `
Array [
  Object {
    "description": "The Green Mile is a 1999 American fantasy drama film writtenand directed by Frank Darabont and based on Stephen Kings 1996 novel of the same name. It stars Tom Hanks as a death row corrections officer during the Great Depression who witnesses supernatural events that occur after an enigmatic inmate (Michael Clarke Duncan) is brought to his facility. David Morse, Bonnie Hunt, Doug Hutchison and James Cromwell appear in supporting roles.The film premiered on December 10, 1999, in the United States to positive reviews from critics, who praised its visual style and performances. It was a commercial success, grossing $286 million from its $60 million budget, and was nominated for four Academy Awards: Best Picture, Best Supporting Actor for Duncan, Best Sound and Best Screenplay Based on Material Previously Produced or Published.",
    "duration": "120",
    "genre": "Drama, Fantasy",
    "id": "-MWJmKyNgxg1U3WKo2MF",
    "img": "https://pbs.twimg.com/media/ELeJbCYXkAE18WI.jpg",
    "rating": "4.9",
    "releaseDate": "1994",
    "title": "Green mile",
  },
]
`;

exports[`getFilteredMovies (filter: "Horror") 1`] = `Array []`;

exports[`getFilteredMovies (filter: "all") 1`] = `
Array [
  Object {
    "description": "The Green Mile is a 1999 American fantasy drama film writtenand directed by Frank Darabont and based on Stephen Kings 1996 novel of the same name. It stars Tom Hanks as a death row corrections officer during the Great Depression who witnesses supernatural events that occur after an enigmatic inmate (Michael Clarke Duncan) is brought to his facility. David Morse, Bonnie Hunt, Doug Hutchison and James Cromwell appear in supporting roles.The film premiered on December 10, 1999, in the United States to positive reviews from critics, who praised its visual style and performances. It was a commercial success, grossing $286 million from its $60 million budget, and was nominated for four Academy Awards: Best Picture, Best Supporting Actor for Duncan, Best Sound and Best Screenplay Based on Material Previously Produced or Published.",
    "duration": "120",
    "genre": "Drama, Fantasy",
    "id": "-MWJmKyNgxg1U3WKo2MF",
    "img": "https://pbs.twimg.com/media/ELeJbCYXkAE18WI.jpg",
    "rating": "4.9",
    "releaseDate": "1994",
    "title": "Green mile",
  },
  Object {
    "description": "Forrest Gump is a 1994 American romantic comedy-drama film directed by Robert Zemeckis and written by Eric Roth. It is based on the 1986 novel of the same name by Winston Groom and stars Tom Hanks, Robin Wright, Gary Sinise, Mykelti Williamson and Sally Field. The story depicts several decades in the life of Forrest Gump (Hanks), a slow-witted but kind-hearted man from Alabama who witnesses and unwittingly influences several defining historical events in the 20th century United States. The film differs substantially from the novel. Principal photography took place between August and December 1993, mainly in Georgia, North Carolina and South Carolina. Extensive visual effects were used to incorporate Hanks into archived footage and to develop other scenes. The soundtrack features songs reflecting the different periods seen in the film.",
    "duration": "125",
    "genre": "Comedy",
    "id": "-MWJof12zk_84pRdWPsq",
    "img": "https://pbs.twimg.com/media/EK7Tf-IXkAErCgW.jpg",
    "rating": "4.3",
    "releaseDate": "1994-12-25",
    "title": "Forrest Gump",
  },
  Object {
    "description": "Brother (Russian: Брат, translit. Brat) is a 1997 Russian neo-noir crime drama film written and directed by Aleksei Balabanov. The film stars Sergei Bodrov Jr. as Danila Bagrov, a young ex-army conscript who becomes embroiled with the Saint Petersburg mob through his criminal older brother. It appeared in the Un Certain Regard section at the 1997 Cannes Film Festival.[1]After its release on VHS in June 1997, Brother unexpectedly became one of the most commercially successful Russian films of the 1990s and quickly rose to cult film status throughout Russia. Due to the films popularity and fan demand, a sequel, Brother 2, was released in 2000",
    "duration": "129",
    "genre": "Thriller",
    "id": "-MWJn2JOvxRpVKrK0Ye3",
    "img": "https://i10.fotocdn.net/s124/a80214fc2d561294/user_xl/2825091898.jpg",
    "rating": "4.5",
    "releaseDate": "1997-06-01",
    "title": "Brother",
  },
]
`;
